Sources/NIO/EventLoop.swift

```diff
@@ -291,6 +295,74 @@ extension TimeAmount: Comparable {
     }
 }
 
+extension TimeAmount {
+    public static func + (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds + rhs.nanoseconds)
+    }
+
+    public static func - (lhs: TimeAmount, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds - rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount.Value, rhs: TimeAmount) -> TimeAmount {
+        return TimeAmount(lhs * rhs.nanoseconds)
+    }
+
+    public static func * (lhs: TimeAmount, rhs: TimeAmount.Value) -> TimeAmount {
+        return TimeAmount(lhs.nanoseconds * rhs)
+    }
+}
+
+/// Represents a point in time.
+///
+/// - note: `Time` should not be used to represent an interval
+public struct Time {
```

Comment:
We can't call this `Time`, it's against our new enhancement policy. At the very least we have to call it `NIOTime`.